easybuild/easyblocks/j/java.py

```diff
@@ -27,13 +27,42 @@
 
 @author: Jens Timmerman (Ghent University)
 """
+
+import os
+import shutil
+
+from distutils.version import LooseVersion
 from easybuild.easyblocks.generic.packedbinary import PackedBinary
+from easybuild.tools.run import run_cmd
 
 
 class EB_Java(PackedBinary):
     """Support for installing Java as a packed binary file (.tar.gz)
     Use the PackedBinary easyblock and set some extra paths.
     """
+    
+    def extract_step(self):
+        """Unpack the source"""
+        if LooseVersion(self.version) < LooseVersion('1.7'):
+            cmd = 'cd %s; bash %s' % (self.builddir, self.src[0]['path'])
```

Comment:
@pescobar please do the `cd` in Python, and there's no need for `bash`, `run_cmd` already runs commands in a subshell

```python
try:
    os.chdir(self.builddir)
except OSError, err:
    raise EasyBuildError("Failed to move to build dir: %s", err)

run_cmd(self.src[0]['path'], log_all=True, simple=True, inp='')
```